heartbeat/monitors/monitor.go

```diff
@@ -121,44 +125,29 @@ func newMonitor(
 		stats:             monitorPlugin.stats,
 	}
 
-	jobs, endpoints, err := monitorPlugin.create(config)
-
-	// Set the correct monitor.id for all jobs
-	var idWrapper func(job Job) Job
 	if m.id != "" {
 		// Ensure we don't have duplicate IDs
 		if _, loaded := uniqueMonitorIDs.LoadOrStore(m.id, m); loaded {
 			return nil, ErrDuplicateMonitorID{m.id}
 		}
-
-		// If they have an explicit ID in the config set it to exactly that
-		idWrapper = func(job Job) Job {
-			return WithMonitorMeta(m.id, m.name, m.pluginName, job)
-		}
 	} else {
 		// If there's no explicit ID generate one
 		hash, err := m.configHash()
 		if err != nil {
 			return nil, err
 		}
-		idWrapper = func(job Job) Job {
-			return WithMonitorMeta(
-				fmt.Sprintf("auto-%s-%#X", mpi.Type, hash),
-				mpi.Name,
-				mpi.Type,
-				job,
-			)
-		}
+		m.id = fmt.Sprintf("auto-%s-%#X", m.typ, hash)
 	}
 
-	jobs = WrapAll(jobs, idWrapper)
-
+	rawJobs, endpoints, err := monitorPlugin.create(config)
+	wrappedJobs := wrappers.WrapCommon(rawJobs, m.id, m.name, m.typ)
```

Comment:
This is the key part of this PR. It is the new point at which all common fields are applied, including timing.